swift/extractor/SwiftOutputRewrite.cpp

```diff
@@ -0,0 +1,329 @@
+#include "SwiftOutputRewrite.h"
+#include "swift/extractor/SwiftExtractorConfiguration.h"
+
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+#include <swift/Basic/OutputFileMap.h>
+#include <swift/Basic/Platform.h>
+#include <unistd.h>
+#include <unordered_set>
+#include <optional>
+#include <iostream>
+
+// Creates a copy of the output file map and updates remapping table in place
+// It does not change the original map file as it is depended upon by the original compiler
+// Returns path to the newly created output file map on success, or None in a case of failure
+static std::optional<std::string> rewriteOutputFileMap(
+    const codeql::SwiftExtractorConfiguration& config,
+    const std::string& outputFileMapPath,
+    const std::vector<std::string>& inputs,
+    std::unordered_map<std::string, std::string>& remapping) {
+  auto newPath = config.tempArtifactDir + '/' + outputFileMapPath;
```

Comment:
please rename `newPath` as it is quite confusing because of the inner loop variable of the same name.